marconi-tutorial/src/Marconi/Tutorial/Indexers/AddressCount.hs

```diff
@@ -228,3 +244,28 @@ mkAddressCountMixedIndexer'
 mkAddressCountMixedIndexer' dbPath keep flush = do
   sqliteIndexer <- mkAddressCountSqliteIndexer dbPath
   pure $ mkMixedIndexer keep flush sqliteIndexer mkListIndexer
+
+-- TEMPORARY --
+-- Trying out some ideas.
```

Comment:
I'm not sure what I think about exposing internal model types directly in the API. The advantage is that avoids a lot of boilerplate (hand-written `aeson` instances, for example). However, the disadvantage is that it couples the internal and external representations, which could in future constrain the implementation if we want to improve the internals without changing the API.

Another consideration, in the general case at least, is that you can end up accidentally exposing private information when a field is added to the internal type. I think it was Slack that was sending (hashed) copies of passwords between workspaces in the traffic between client and server. Someone I worked with at the time pointed out that this would have been impossible if the application had used separate internal and "edge" types, because no-one is going to put a field called "password" in an edge type.

Since our present customers are all internal, these disadvantages probably don't matter too much at the moment, but presumably the situation could change in the future.